### ISSUE TITLE
Ordered lists with embedded codeblocks or quotations not working

### ISSUE THREAD

**Issue by marcofranssen** [markdownlint/markdownlint#323]
## Description

Using ordered lists with embedded codeblocks give errors on markdown.

### Environment

Github actions using the action `actionshub/markdownlint@1.2.2`.

### Expected Behavior

The markdown is seen as valid markdown.

### Actual Behavior

MD029 error on `2. Second step`
Same happens if you add quotations or combinations of this with code blocks.

### Following markdown:

```md
# Example ordered

1. First step

    To do abc you can run following code.

    ```bash
    echo "Step 1"
    echo "Hello World"
    ```

2. Second step

    Now you can run the example as defined here.

3. Third step

     Enjoying markdown linter.
```

### Renders as:

# Example ordered

1. First step

    To do abc you can run following code.

    ```bash
    echo "Step 1"
    echo "Hello World"
    ```

2. Second step

    Now you can run the example as defined here.

    > Looks also quotations will give errors

3. Third step

     Enjoying markdown linter.
